src/main/java/edu/ucla/library/iiif/fester/handlers/CheckEndpointsHandler.java

```diff
@@ -23,54 +26,72 @@
 
         Future.future(put -> {
             myS3Client.put(myS3Bucket, UPLOAD_KEY, Buffer.buffer(upload.toJSON().encodePrettily()), putResponse -> {
-                final int statusCode = putResponse.statusCode();
-                final String statusMessage = putResponse.statusMessage();
-                endpoints.put(Status.PUT_RESPONSE, statusCode);
-                determineEndpointStatus(statusCode, statusMessage, endpoints, put);
+                if (putResponse.failed()) {
+                    final UnexpectedStatusException error = (UnexpectedStatusException) putResponse.cause();
+                    final int statusCode = error.getStatusCode();
+                    final String message = error.getMessage();
+
+                    determineEndpointStatus(Status.PUT_RESPONSE, statusCode, message, endpoints, put);
+                } else {
+                    determineEndpointStatus(Status.PUT_RESPONSE, HTTP.OK, Status.OK, endpoints, put);
+                }
             });
         }).compose(addGet -> {
             return Future.future(get -> {
                 myS3Client.get(myS3Bucket, UPLOAD_KEY, getResponse -> {
-                    final int statusCode = getResponse.statusCode();
-                    final String statusMessage = getResponse.statusMessage();
-                    endpoints.put(Status.GET_RESPONSE, statusCode);
-                    determineEndpointStatus(statusCode, statusMessage, endpoints, get);
+                    if (getResponse.failed()) {
+                        final UnexpectedStatusException error = (UnexpectedStatusException) getResponse.cause();
+                        final int statusCode = error.getStatusCode();
+                        final String message = error.getMessage();
+
+                        determineEndpointStatus(Status.GET_RESPONSE, statusCode, message, endpoints, get);
+                    } else {
+                        determineEndpointStatus(Status.GET_RESPONSE, HTTP.OK, Status.OK, endpoints, get);
+                    }
                 });
-            } );
+            });
         }).compose(addDel -> {
             return Future.future(delete -> {
                 myS3Client.delete(myS3Bucket, UPLOAD_KEY, deleteResponse -> {
-                    final int statusCode = deleteResponse.statusCode();
-                    final String statusMessage = deleteResponse.statusMessage();
-                    endpoints.put(Status.DELETE_RESPONSE, statusCode);
-                    determineEndpointStatus(statusCode, statusMessage, endpoints, delete);
+                    if (deleteResponse.failed()) {
+                        final UnexpectedStatusException error = (UnexpectedStatusException) deleteResponse.cause();
+                        final int statusCode = error.getStatusCode();
+                        final String message = error.getMessage();
+
+                        determineEndpointStatus(Status.GET_RESPONSE, statusCode, message, endpoints, delete);
```

Comment:
Should this refer to `Status.DELETE_RESPONSE` instead?

Reply:
Yep. Fixing.